File: LeetCode/0003.longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int pos[256];
        memset(pos,-1,sizeof(pos));
        int st = 0,ed = 0,res = 0;
        int len = s.size();
        for (int i = 0;i < len;i++){
        	int index = s[i];
        	if (pos[index] == -1 || pos[index] < st){
        		pos[index] = i;
        		res = max(res,i - st + 1);
        	}else{
        		st = pos[index] + 1;
				pos[index] = i; 
        	}
        }
        return res;
    }
};

File: LeetCode/0003.longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.kt
import kotlin.math.max

class Solution {
    fun lengthOfLongestSubstring(s: String): Int {
        val pos = IntArray(256)
        pos.fill(-1, 0, pos.size)
        var st = 0
        var res = 0
        for (i in s.indices) {
            val index = s[i].toInt()
            if (pos[index] == -1 || pos[index] < st) {
                res = max(res, i - st + 1)
            } else {
                st = pos[index] + 1
            }
            pos[index] = i
        }
        return res
    }
}